go/fn/subobject.go

```diff
@@ -35,3 +36,55 @@ func (o *SubObject) GetMap(k string) *SubObject {
 	}
 	return &SubObject{obj: m}
 }
+
+// GetSlice gets the component in the `fields` path. It expects the component to be
+// a slice of SubObjects.
+func (o *SubObject) GetSlice(fields ...string) (SliceSubObjects, bool, error) {
+	var mapVariant *internal.MapVariant
+	if len(fields) > 1 {
+		m, found, err := o.obj.GetNestedMap(fields[:len(fields)-1]...)
+		if err != nil || !found {
+			return nil, found, err
+		}
+		err = m.Node().Decode(mapVariant)
+		return nil, true, err
```

Comment:
Did you mean to return here?  Maybe only if there's an error?

Reply:
Good catch!